src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -205,6 +208,17 @@ internal bool IsCPSProject(CodeAnalysis.Project project)
             return false;
         }
 
+        protected override bool CanApplyCompilationOptionChange(CompilationOptions oldOptions, CompilationOptions newOptions, CodeAnalysis.Project project)
+        {
+            if (!(oldOptions is CSharpCompilationOptions oldCSharpOptions))
```

Comment:
why do  we have a 'is' check for the oldOptions but not the new options?

Reply:
because i assume you can't change the language of a project